src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs

```diff
@@ -10,100 +9,261 @@
-                    initializers.Add(new GraphInitializerInfo(GraphInitializerType.InitializeMethod, node, symbol));
+                    var delegateInfo = del.Base == null
+                        ? new DataViewDelegateInfo(del.Item.Node, del.Item.Symbol)
+                        : new DataViewDelegateInfo(del.Item.Node, del.Item.Symbol, new DataViewDelegateInfo(del.Base.Item.Node, del.Base.Item.Symbol));
+
+                    viewDelegateByNameDictionary[del.Item.Symbol.Name] = delegateInfo;
                 }
             }
 
-            Initializers = initializers.ToImmutableArray();
+            return viewDelegateByNameDictionary.ToImmutableDictionary();
         }
 
+		private ImmutableDictionary<string, ActionInfo> GetActions()
+		{
+			if (Type == GraphType.None)
+				return ImmutableDictionary.Create<string, ActionInfo>(StringComparer.OrdinalIgnoreCase);
+
+			var systemActionsRegister = new PXSystemActions.PXSystemActionsRegister(_pxContext);
+			var rawActionInfos = Type == GraphType.PXGraph
+				? Symbol.GetActionSymbolsWithTypesFromGraph(_pxContext)
+				: Symbol.GetActionsFromGraphExtensionAndBaseGraph(_pxContext);
+
+			return rawActionInfos.ToLookup(a => a.Item.ActionSymbol.Name, StringComparer.OrdinalIgnoreCase)
+								 .ToImmutableDictionary(group => group.Key,
+														group => CreateActionInfo(group.First()),
+														keyComparer: StringComparer.OrdinalIgnoreCase);
+
+			//--------------------------------------------------------Local Function--------------------------------------------
+			ActionInfo CreateActionInfo(GraphOverridableItem<(ISymbol ActionSymbol, INamedTypeSymbol ActionType)> item)
+			{
+				var (actionSymbol, actionType) = item.Item;
+
+				ActionInfo baseActionInfo = item.Base != null
+					? baseActionInfo = CreateActionInfo(item.Base)
+					: null;
+
+				return baseActionInfo == null
+					? new ActionInfo(actionSymbol, actionType, systemActionsRegister.IsSystemAction(actionType))
+					: new ActionInfo(actionSymbol, actionType, systemActionsRegister.IsSystemAction(actionType), baseActionInfo);
+			}
+		}
+
+		private ImmutableDictionary<string, ActionHandlerInfo> GetActionHandlers()
+		{
+			if (Type == GraphType.None)
+				return ImmutableDictionary.Create<string, ActionHandlerInfo>(StringComparer.OrdinalIgnoreCase);
+
+			var rawActionHandlerInfos = Type == GraphType.PXGraph
+				? Symbol.GetActionHandlersFromGraph(ActionsByNames, _pxContext, _cancellation)
+				: Symbol.GetActionHandlersFromGraphExtensionAndBaseGraph(ActionsByNames, _pxContext, _cancellation);
+
+			return rawActionHandlerInfos.ToLookup(handler => handler.Item.Symbol.Name, StringComparer.OrdinalIgnoreCase)
+										.ToImmutableDictionary(group => group.Key,
+															   group => CreateActionHandlerInfo(group.First()),
+															   keyComparer: StringComparer.OrdinalIgnoreCase);
+
+			//--------------------------------------------------------Local Function--------------------------------------------
```

Comment:
I think this comment looks ugly. What do you think, can we just keep local functions in the end of methods without any comment?

Reply:
ok